library/core/src/num/int_macros.rs

```diff
@@ -510,9 +510,15 @@ macro_rules! int_impl {
         #[inline(always)]
         #[cfg_attr(miri, track_caller)] // even without panics, this helps for Miri backtraces
         pub const unsafe fn unchecked_add(self, rhs: Self) -> Self {
-            // SAFETY: the caller must uphold the safety contract for
-            // `unchecked_add`.
-            unsafe { intrinsics::unchecked_add(self, rhs) }
+            debug_assert_nounwind!(
+                !self.overflowing_add(rhs).1,
+                concat!(stringify!($SelfT), "::unchecked_add cannot overflow"),
+            );
+            // SAFETY: this is guaranteed to be safe by the caller.
+            unsafe {
+                let lhs = self;
```

Comment:
This extra `let` is going to cause more LLVM IR to be generated in order to have debuginfo for it. I learned about this unfortunate behavior only recently: https://github.com/rust-lang/rust/blob/8f359beca4e58bc3ae795a666301a8f47023044c/library/core/src/intrinsics.rs#L2688-L2691

I wish I had a way to tell rustc that we don't need variable-level debuginfo for a span of code, but here we are.

Reply:
Ugh, I didn't realise it got all the way to the LLVM IR level. It's unfortunate, but I'll remove the extra lets from these operations.